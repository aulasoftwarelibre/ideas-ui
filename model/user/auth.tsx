export interface Session {
  user?: User;
}

export interface User {
  name?: string;
  email?: string;
  image?: string;
}
