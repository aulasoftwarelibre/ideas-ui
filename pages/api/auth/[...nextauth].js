import NextAuth from 'next-auth';
import Providers from 'next-auth/providers';

const options = {
  // Configure one or more authentication providers
  providers: [
    // TODO: Load this only in dev
    Providers.Credentials({
      name: "Credentials",
      session: {
        jwt: true,
      },
      credentials: {
        username: { label: "Usuario", type: "text" },
        password: { label: "Clave", type: "password" },
      },
      authorize: async (credentials) => {
        const user = {
          id: 1,
          name: "J Smith",
          email: "jsmith@example.com",
          image: "/ideas-isotipo-naranja.svg",
        };

        if (user) {
          return Promise.resolve(user);
        } else {
          return Promise.resolve(null);
        }
      },
    }),
  ],

  // A database is optional, but required to persist accounts in a database
  database: process.env.DATABASE_URL,
};

export default (req, res) => NextAuth(req, res, options);
