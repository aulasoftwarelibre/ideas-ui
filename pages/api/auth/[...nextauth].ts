import { NextApiRequest, NextApiResponse } from 'next'
import NextAuth, { InitOptions } from 'next-auth'
import Providers from 'next-auth/providers'

const options = {
  session: {
    jwt: true,
  },
  // Configure one or more authentication providers
  providers: [
    // TODO: Load this only in dev
    Providers.Credentials({
      name: 'Credentials',
      credentials: {
        username: { label: 'Usuario', type: 'text' },
        password: { label: 'Clave', type: 'password' },
      },
      authorize: async (_credentials) => {
        const user = {
          id: 1,
          name: 'J Smith',
          email: 'jsmith@example.com',
          image: '/avatar.svg',
        }

        if (user) {
          return Promise.resolve(user)
        } else {
          return Promise.resolve(null)
        }
      },
    }),
  ],

  // A database is optional, but required to persist accounts in a database
  database: process.env.DATABASE_URL,
} as InitOptions

export default (req: NextApiRequest, res: NextApiResponse): Promise<void> =>
  NextAuth(req, res, options)
