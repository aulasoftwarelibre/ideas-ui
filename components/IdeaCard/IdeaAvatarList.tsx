import React from "react";
import { User } from "../../model/user/auth";

export interface Props {
  attendess: User[];
}

export const IdeaAvatarList: React.FunctionComponent<Props> = ({
  attendess,
}) => {
  const avatars = attendess
    .slice(0, 4)
    .map((attendee, index) => (
      <img
        className="-ml-1 first:ml-0 inline-block h-8 w-8 rounded-full text-white shadow-solid"
        src={attendee.image ?? "/avatar.svg"} 
        alt=""
        key={index}
      />
    ));
  return (
    <div className="flex overflow-hidden p-1">
      {avatars}
      <div className="font-bold text-white text-base ml-3 mt-1">
        {attendess.length}
      </div>
    </div>
  );
};

export default IdeaAvatarList;
