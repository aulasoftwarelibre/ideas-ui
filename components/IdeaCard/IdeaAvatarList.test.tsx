import { shallow } from "enzyme";
import React from "react";
import { User } from "../../model/user/auth";
import { IdeaAvatarList } from "./IdeaAvatarList";

describe("AvatarList", () => {

  const attendess: User[] = [{}, {}, {}];

  const component = shallow(
    <IdeaAvatarList attendess={attendess}></IdeaAvatarList>
  );

  it("should be rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });
});