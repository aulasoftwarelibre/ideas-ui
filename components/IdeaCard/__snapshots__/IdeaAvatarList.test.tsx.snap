// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarList should be rendered correctly 1`] = `ShallowWrapper {}`;
